MoreLinq/Identity.cs

```diff
@@ -0,0 +1,29 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2022 Turning Code, LLC. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    public static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Returns the identity function for a given type.
+        /// </summary>
+        /// <typeparam name="T">The type of identity function</typeparam>
+        /// <returns>A reference to the identity function</returns>
+        public static T Identity<T>(T x) => x;
```

Comment:
Yes, the advantages of compiling the `Identity<T>` method once should be available to consumers if desired.

Reply:
The identity function is already a micro optimization and while it's fair to use within the library, I don't think offering it as a public API is worth committing to it. It would be the first stand-alone helper function that could open a can of worms. Ideally this is something that should be part of .NET proper. We can keep it private for now and always review the decision.